packages/server/validators/events.js

```diff
@@ -0,0 +1,229 @@
+    .withMessage("Title cannot be null or empty")
+    .isLength({ min: 1, max: 255 })
+    .withMessage("Title length must be between 1 to 255 characters"),
+
+  //Description checks
+  body("description", "Invalid description")
+    .trim()
+    .escape()
+    .isLength({ max: 255 })
+    .withMessage("Description max length is 255 characters")
+    .exists({ checkFalsy: true })
+    .withMessage("Description cannot be null or empty"),
+
+  //Location checks
+  body("location", "Invalid location")
+    .trim()
+    .escape()
+    .exists({ checkFalsy: true })
+    .withMessage("location cannot be null or empty")
+    .isLength({ max: 255 })
+    .withMessage("Location max length is 255 characters"),
+
+  //Thumbnail checks
+  body("thumbnail", "Invalid thumbnail")
+    .trim()
+    .escape()
+    .exists({ checkFalsy: true })
+    .withMessage("thumbnail cannot be null or empty")
+    .isLength({ max: 255 })
+    .withMessage("Thumbnail max length is 255 characters")
+    .custom(async (value) => {
+      if (!/\.(jpg|png|JPG|PNG|JPEG|jpeg)$/.test(value)) {
+        throw new Error("Thumbnail is not a valid image file");
+      }
+    }),
+
+  //Start Date checks
+  //Convert end date to a Date object for easier comparison
+  check("endDate").toDate(),
+  body("startDate", "Invalid dates")
+    .trim()
+    .escape()
+    .exists({ checkFalsy: true })
+    .withMessage("start date cannot be null or empty")
+    .isISO8601()
+    .toDate()
+    .withMessage("Start date is not in a valid date format")
+    .custom(async (startDate, { req }) => {
+      //If new end date is provided, check if start date is before the end date
+      if (req.body.endDate) {
+        if (startDate > req.body.endDate) {
+          throw new Error("Start date is after new end date");
+        }
+      }
+    }),
+
+  //End Date checks
+  //Similar checks to start sate but makes sure end date is after the start
+  check("startDate").toDate(),
+  body("endDate", "Invalid end date")
+    .trim()
+    .escape()
+    .exists({ checkFalsy: true })
+    .withMessage("end date cannot be null or empty")
+    .isISO8601()
+    .toDate()
+    .withMessage("End date is not in a valid date format")
+    .custom(async (endDate, { req }) => {
+      if (req.body.startDate) {
+        if (endDate < req.body.startDate) {
+          throw new Error("End date is before new start date");
+        }
+      }
+    }),
+];
+
+const updateEventValidator = [
+  //Event Id checks
+  param("eventId", "Invalid event ID")
+    .trim()
+    .escape()
+    .isUUID()
+    .withMessage("Not a UUID")
+    .exists({ checkFalsy: true })
+    .withMessage("Event Id cannot be null or empty")
+    .custom(async (value) => {
+      try {
+        await EventController.getEvent(value); //Check if event exists by getting it
+      } catch (error) {
+        //If event does not exists, then a NotFoundError is thrown from controller
+        throw new Error(error);
+      }
+    }),
```

Comment:
This is a duplicate check from the `eventIdValidator` already defined at the bottom of the page. You can get rid of this and instead use both validators in the update event route handler. It would look something like this
```js
router.put(
  "/:eventId",
  AuthController.authenticate,
  eventIdValidator,
  updateEventValidator,
  async (request, response) => {
  //  ...
});
```